src/modules/Packages.rb

```diff
@@ -2490,7 +2511,7 @@ def patterns_to_install
     end
 
     def report_missing_pattern(pattern_name)
-      if optional_default_patterns.include?(pattern_name)
+      if (optional_default_patterns | resolvable_optional_patterns).include?(pattern_name)
```

Comment:
It should additionally check that the pattern is *not* listed as mandatory as well. The same pattern might be added by more components with different requirements (optional/mandatory).

If one component adds the pattern as optional and another component sets the same pattern as mandatory then the error should be still reported. With this code the pattern becomes optional globally for all components.

Reply:
It would be nice to add a test case for this...

Reply:
👍 I was currently checking exactly that, and was not sure if it should always prevail the last one at least for products (installation.xml) vs control file (control.xml). I mean, if control file defines it as default_pattern but product as optional_default pattern if it should be only as optional or in mandatory and optional lists.

After a discussion we decided the Resolvables will be just added and we should check if them are mandatory.